activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb

```diff
@@ -77,7 +77,7 @@ def cache_sql(sql, binds)
         result =
           if @query_cache[sql].key?(binds)
             ActiveSupport::Notifications.instrument("sql.active_record",
-              :sql => sql, :binds => binds, :name => "CACHE", :connection_id => object_id)
+              :sql => sql.freeze, :binds => binds.freeze, :name => "CACHE", :connection_id => object_id)
```

Comment:
Yeah, mutating the values on the query cache is another problem for sure. I didn't want to fix it here, tho, as that part I am not sure how to implement yet.